js_modules/dagit/packages/core/src/instance/backfill/BackfillPage.tsx

```diff
@@ -0,0 +1,310 @@
+import {RunFilterToken, runsPathWithFilters} from '../../runs/RunsFilterInput';
+
+import {
+  BackfillStatusesByAssetQuery,
+  BackfillStatusesByAssetQueryVariables,
+} from './types/BackfillPage.types';
+
+dayjs.extend(duration);
+dayjs.extend(relativeTime);
+
+export const BackfillPage = () => {
+  const {backfillId} = useParams<{backfillId: string}>();
+  useTrackPageView();
+  useDocumentTitle(`Backfill | ${backfillId}`);
+
+  const {data, loading} = useQuery<
+    BackfillStatusesByAssetQuery,
+    BackfillStatusesByAssetQueryVariables
+  >(BACKFILL_DETAILS_QUERY, {
+    variables: {backfillId},
+  });
+
+  console.log({backfillId, data, loading});
+
+  const backfill = data?.partitionBackfillOrError;
+
+  function content() {
+    if (!backfill || loading) {
+      return (
+        <Box padding={64}>
+          <Spinner purpose="page" />
+        </Box>
+      );
+    }
+    if (backfill.__typename === 'PythonError') {
+      return <PythonErrorInfo error={backfill} />;
+    }
+
+    function getRunsUrl(status: 'requested' | 'complete' | 'failed' | 'targeted') {
+      const filters: RunFilterToken[] = [
+        {
+          token: 'tag',
+          value: `dagster/backfill=${backfillId}`,
+        },
+      ];
+      switch (status) {
+        case 'requested':
+          filters.push(
+            {
+              token: 'status',
+              value: RunStatus.STARTED,
+            },
+            {
+              token: 'status',
+              value: RunStatus.QUEUED,
+            },
+            {
+              token: 'status',
+              value: RunStatus.STARTING,
+            },
+          );
+          break;
+        case 'complete':
+          filters.push({
+            token: 'status',
+            value: RunStatus.SUCCESS,
+          });
+          break;
+        case 'failed':
+          filters.push({
+            token: 'status',
+            value: RunStatus.FAILURE,
+          });
+          filters.push({
+            token: 'status',
+            value: RunStatus.CANCELED,
+          });
+          break;
+      }
+      return runsPathWithFilters(filters);
+    }
+
+    return (
+      <>
+        <Box
+          padding={24}
+          flex={{
+            direction: 'row',
+            justifyContent: 'space-between',
+            wrap: 'nowrap',
+            alignItems: 'center',
+          }}
+        >
+          <Detail label="Created" detail="Mar 22, 5:00 PM" />
+          <Detail
+            label="Duration"
+            detail={
+              <Duration
+                start={backfill.timestamp * 1000}
+                end={backfill.endTimestamp ? backfill.endTimestamp * 1000 : null}
+              />
+            }
+          />
+          {/* <Detail label="Partition Selection" detail="test" /> */}
+          <Detail label="Status" detail={<StatusLabel status={backfill.status} />} />
+        </Box>
+        <Table $monospaceFont={false}>
```

Comment:
This all looks great, but it seems asset specific and there isn't a fallback for non-asset backfills. I think that the updates to BackfillMessaging.tsx and BackfillRow make this page reachable for both asset backfills AND op job backfills, and the content here might not make sense for the latter. Maybe we could make this table conditional? Would be good to see what happens when you make the GraphQL query for `assetPartitionsStatusCounts` on a non-asset backfill

Reply:
Good call out.

How do I create a non-asset backfill? Is there a toy I can use?